### PR TITLE
feat: add json output to password policy manager

### PR DESCRIPTION
**Pull request by GaryJBlake**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

- Added `-json` switch to each `Publish-` function
- Refactored `Invoke-PasswordPolicyManager `to either generate the report or generate the JSON

Signed-off-by: Gary Blake <gblake@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is something else.
      Please describe:

**Context of the Pull Request***

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -37451,6 +37568,2386 @@ Export-ModuleMember -Function Test-NtpServer
 #######################################################################################################################
 #Region                                D E P R E C A T E D  F U N C T I O N S                               ###########
 
+Function Join-ESXiActiveDirectory {
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:37571]: Looks like the functions removed in #186 slipped back in.

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Looks like the functions removed in https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/pull/186 slipped back in.

File: PowerValidatedSolutions.psm1 (hunk 1 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1]: Fixed, good catch that's my bad

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀

**Pull request merged**